### ISSUE TITLE
召回率太低吗？

### ISSUE THREAD

**Issue by RichieMay** [AITTSMD/MTCNN-Tensorflow#141]
我在你的代码中添加了一个计算召回率的方法如下：
def cale_recall(cls_prob,label):
pred = tf.argmax(cls_prob,axis=1)
    label_int = tf.cast(label,tf.int64)
    cond = tf.where(tf.greater(label_int,0))  # 获取正样本的在label中索引位置
    picked = tf.squeeze(cond)
    label_picked = tf.gather(label_int,picked)  #获取正样本的标签
    pred_picked = tf.gather(pred,picked)  # 获取应该预测为正样本的预测值
    recall_op = tf.reduce_mean(tf.cast(tf.equal(label_picked, pred_picked),tf.float32))
    return recall_op
该方法添加在PNet计算accuracy的下一行

然后我的数据集中没有landmark，因此在计算loss的时候去掉landmark_loss后如下：
train_op, lr_op = train_model(base_lr, radio_cls_loss * cls_loss_op + radio_bbox_loss * bbox_loss_op + L2_loss_op, num) 

训练的时候，准确度在94%左右，召回率大多分布在80多，这样算正常吗？数据集用的WiderFaces 生成的样本数，负：正：部分=589227:196409:196409 = 3:1:1。

环境配置：
tensorflow 1.8
python 3.6.5
![recall](https://user-images.githubusercontent.com/29646610/41828243-d56e9a26-7866-11e8-9ec3-f8de3316c0b3.png)

**Comment by AITTSMD** [comment]
@RichieMay 这样测试的是一个batchsize的召回率吧？我没这样测过，不知道算是高还是低，感觉有点低。你把训练好的模型在fddb上测试一下召回率。

**Comment by RichieMay** [comment]
批次召回率都这么低 我在想加起来也不会好到哪里去 训练的时候就评估 没想到这么低 我自己也实现了一遍 召回率也低 是不是说明训练的效果其实不怎么样呢

**Comment by RichieMay** [comment]
对的，批次召回率低也就意味着总召回率也低，那么召回率这么低是否意味着训练的效果其实并不好？

**Comment by RichieMay** [comment]
@AITTSMD 对的，批次召回率低也就意味着总召回率也低，那么召回率这么低是否意味着训练的效果其实并不好？

**Comment by NoahLu1105** [comment]
@RichieMay  针对召回率低的问题，有啥建议不？
